packages/block-library/src/embed/index.js

```diff
@@ -163,39 +166,26 @@ export function getEmbedEdit( title, icon ) {
 
 			if ( iframe && iframe.height && iframe.width ) {
 				const aspectRatio = ( iframe.width / iframe.height ).toFixed( 2 );
-				let aspectRatioClassName;
-
-				switch ( aspectRatio ) {
+				const aspectRatios = [
 					// Common video resolutions.
-					case '2.33':
-						aspectRatioClassName = 'wp-embed-aspect-21-9';
-						break;
-					case '2.00':
-						aspectRatioClassName = 'wp-embed-aspect-18-9';
-						break;
-					case '1.78':
-						aspectRatioClassName = 'wp-embed-aspect-16-9';
-						break;
-					case '1.33':
-						aspectRatioClassName = 'wp-embed-aspect-4-3';
-						break;
+					{ ratio: '2.33', className: 'wp-embed-aspect-21-9' },
+					{ ratio: '2.00', className: 'wp-embed-aspect-18-9' },
+					{ ratio: '1.78', className: 'wp-embed-aspect-16-9' },
+					{ ratio: '1.33', className: 'wp-embed-aspect-4-3' },
 					// Vertical video and instagram square video support.
-					case '1.00':
-						aspectRatioClassName = 'wp-embed-aspect-1-1';
-						break;
-					case '0.56':
-						aspectRatioClassName = 'wp-embed-aspect-9-16';
-						break;
-					case '0.50':
-						aspectRatioClassName = 'wp-embed-aspect-1-2';
-						break;
-				}
-
-				if ( aspectRatioClassName ) {
-					return {
-						[ aspectRatioClassName ]: allowResponsive,
-						'wp-has-aspect-ratio': allowResponsive,
-					};
+					{ ratio: '1.00', className: 'wp-embed-aspect-1-1' },
+					{ ratio: '0.56', className: 'wp-embed-aspect-9-16' },
+					{ ratio: '0.50', className: 'wp-embed-aspect-1-2' },
+				];
+				// Given the actual aspect ratio, find the widest ratio to support it.
+				for ( let idx = 0; idx < aspectRatios.length; idx++ ) {
```

Comment:
What's the `idx` stand for?

Also, any reason not to use `for... in/of`?

Reply:
>Also, any reason not to use `for... in/of`?

Related: https://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea

In this case, guaranteed iteration order seems important.

Reply:
Heh, good point 👍

Reply:
idx was short for index, I thought it was one of the commonly used index var names (i, and x being others) and I hope the usage is obvious from the code.  We use this in other places too, but I'll change it to `ratioIndex`

Reply:
The usage is pretty obvious, but I think I have a pet peeve regarding brief variable names. Sorry for the nitpicking!